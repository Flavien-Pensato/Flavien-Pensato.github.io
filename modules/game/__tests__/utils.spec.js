import {
  debug, getCanvas, loadSprite, addElement,
} from '../utils';

jest.unmock('../utils.js');

describe('utils', () => {
  describe('getCanvas', () => {
    it('should return canvas if exist', () => {
      expect(getCanvas('canvasId')).toBeDefined();
    });

    it('should return nothing and log the error', () => {
      expect(getCanvas('badCanvasId')).toBeUndefined();
      expect(debug).toHaveBeenCalledTimes(1);
    });
  });

  describe('loadSprite', () => {
    it('should load my image', async () => {
      expect.assertions(1);

      try {
        const sprite = await loadSprite('src/sprite.png');

        expect(sprite).toBeDefined();
      } catch (error) {
        throw new Error(error);
      }
    });

    it('should throw error when image not found', async () => {
      expect.assertions(1);

      try {
        await loadSprite('src/bad_sprite.png');
      } catch (error) {
        expect(error).toBeInstanceOf(Error);
      }
    });
  });


  describe('addElement', () => {
    it('should add my element with his sprite', async () => {
      const element = await addElement({ name: 'toto' }, 'src/sprite.png');
      expect(element).toMatchSnapshot();
    });

    it('should fail because of sprite and return null', async () => {
      const element = await addElement({ name: 'toto' }, 'src/bad_sprite.png');

      expect(element).toBeNull();
      expect(debug).toHaveBeenCalledWith('No Element added. Missing `Sprite`.');
    });

    it('should fail because of element and return null', async () => {
      const element = await addElement(undefined, 'src/sprite.png');

      expect(element).toBeNull();
      expect(debug).toHaveBeenCalledWith('No Element added. Missing `Element`.');
    });
  });
});
