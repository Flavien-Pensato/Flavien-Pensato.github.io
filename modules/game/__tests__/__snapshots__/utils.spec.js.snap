// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils addElement should add my element with his sprite 1`] = `
Object {
  "Sheight": 27,
  "Sprite": <img />,
  "Swidth": 14,
  "Sx": 10,
  "Sy": 5,
  "X": 32,
  "Y": 60,
  "collisionCaseTypes": Array [
    "brick",
    "ground",
    "question",
    "pipe",
  ],
  "collisionLeft": 5,
  "collisionRight": -10,
  "direction": "right",
  "gravity": Object {
    "max": 0.35,
    "speed": 0.02,
    "value": 0,
  },
  "height": 54,
  "motion": Object {
    "max": 0.3,
    "speed": 0.008,
    "value": 0,
  },
  "name": "toto",
  "onGround": true,
  "width": 28,
}
`;
